tests/test_object_model/test_site.py

```diff
@@ -1,99 +1,321 @@
+            "msl": {"value": 0.890, "units": "meters"},
+        },
+        "obs_point": [
+            {
+                "name": "ashley_river",
+                "description": "Ashley River - James Island Expy",
+                "lat": 32.7765,
+                "lon": -79.9543,
+            },
+            {
+                "name": 8665530,
+                "description": "Charleston Cooper River Entrance",
+                "ID": 8665530,
+                "lat": 32.78,
+                "lon": -79.9233,
+            },
+        ],
+        "benefits": {
+            "current_year": 2023,
+            "current_projection": "current",
+            "baseline_strategy": "no_measures",
+            "event_set": "test_set",
+        },
+        "scs": {"file": "scs_rainfall.csv", "type": "type_3"},
+        "standard_objects": {...},
+    }
+    return config_values
 
-def test_read_site_toml(test_db):
-    test_toml = test_database / "charleston" / "static" / "site" / "site.toml"
 
-    assert test_toml.is_file()
+@pytest.fixture
+def test_tomls(test_db):
+    toml_files = [
+        test_db.static_path / "site" / "site.toml",
+        test_db.static_path / "site" / "site_without_river.toml",
+    ]
+    return toml_files
 
-    test_data = Site.load_file(test_toml)
-    assert isinstance(test_data.attrs.name, str)
-    assert isinstance(test_data.attrs.sfincs, SfincsModel)
-    assert isinstance(test_data.attrs.dem, DemModel)
-    assert isinstance(test_data.attrs.obs_station, Obs_stationModel)
-    assert isinstance(test_data.attrs.water_level, WaterLevelReferenceModel)
-    assert isinstance(test_data.attrs.water_level.other[0].height, UnitfulLength)
-    assert test_data.attrs.lat == 32.77
-    assert test_data.attrs.slr.vertical_offset.value == 0.6
-    assert test_data.attrs.fiat.exposure_crs == "EPSG:4326"
-    assert test_data.attrs.river[0].mean_discharge.value == 5000
 
+@pytest.fixture
+def test_sites(test_db, test_tomls):
+    test_sites = {toml_file.name: Site.load_file(toml_file) for toml_file in test_tomls}
+    return test_sites
 
-def test_read_site_toml_without_river(test_db):
-    test_toml = (
-        test_database / "charleston" / "static" / "site" / "site_without_river.toml"
-    )
 
-    assert test_toml.is_file()
+def test_loadFile_validFiles(test_tomls):
+    for toml_filepath in test_tomls:
+        Site.load_file(toml_filepath)
 
-    test_data = Site.load_file(test_toml)
 
-    assert isinstance(test_data.attrs.name, str)
-    assert isinstance(test_data.attrs.sfincs, SfincsModel)
-    assert isinstance(test_data.attrs.dem, DemModel)
-    assert isinstance(test_data.attrs.obs_station, Obs_stationModel)
-    assert test_data.attrs.lat == 32.77
-    assert test_data.attrs.slr.vertical_offset.value == 0.6
-    assert test_data.attrs.fiat.exposure_crs == "EPSG:4326"
+def test_loadFile_invalidFile_raiseFileNotFoundError(test_db):
+    with pytest.raises(FileNotFoundError):
+        Site.load_file("not_a_file.toml")
```

Comment:
Maybe also assert on the error message if you specify that?

Reply:
We dont customize this error. It is raised by open(). Ive added it anyways